File: Python/stack.py
''' 
        author : Prabhat kumar
        date : 28 jan 2020


Design, Develop and Implement a menu driven Program for the following operations on
STACK of Integers (Array Implementation of Stack with maximum size MAX)
a. Push an Element on to Stack
b. Pop an Element from Stack
c. Demonstrate how Stack can be used to check Palindrome
d. Demonstrate Overflow and Underflow situations on Stack
Display the status of Stack
f. Exit
Support the program with appropriate functions for each of the above operations
'''

from os import system
class Stack:
    def __init__(self,size):
        self.stack=[]
        self.top=-1
        self.capacity=size
    def display(self):
        print("Capacity : ",self.capacity)
        print("Filled  : ",self.top+1)
        for i in range(self.top,-1,-1):
            print()
            print("|",self.stack[i],"|\n--")
    def push(self):
        if self.top==self.capacity:
            print("Overflow")
        else:
            data=int(input("Enter number push in stack : "))
            self.stack.append(data)
            self.top+=1
    def pop(self):
        if self.top==-1:
            print("Underflow")
        else:
            return self.arr.pop()
if __name__=="__main__":
    system("clear")
    size=int(input("Enter size of the stack "))
    stack=Stack(size)
    while True:
        system("clear")
        print("\tChoose the options ")
        print("\t1. Push element in stack")
        print("\t2. Pop element from stack")
        print("\t3. Show current stats of stack")
        print("\t4. Exit")
        options=int(input("Choose from 1-4 : "))
        if options==1:
            stack.push()
        elif options==2:
            stack.pop()
        elif options==3:
            stack.display()
        elif options==4:
            break
        else:
            print("Choose right option...")
        input("Press enter to continue...")

File: C++/Stack.cpp
/*
              author : Prabhat kumar
              date : 25 jan 2020
              
Design, Develop and Implement a menu driven Program for the following operations on
STACK of Integers (Array Implementation of Stack with maximum size MAX)
a. Push an Element on to Stack
b. Pop an Element from Stack
c. Demonstrate how Stack can be used to check Palindrome
d. Demonstrate Overflow and Underflow situations on Stack
Display the status of Stack
f. Exit


Note : for window cls is used for clear screen
       so while running this program on unix or
       unix style oprating system change cls to
       clear
*/
#include <stdio.h>
#include <iostream>
#include <stdlib.h>
#include <conio.h>
using namespace std;
class Stack
{
public:
  int *stack;
  int capacity;
  int TOP;
    Stack (int size) {
    stack = new int[size];
      capacity = size;
      TOP = -1;
  }
  void print (){
	cout<<endl;
    if (TOP == -1)
      cout << "\tstack is empty\n";
    for (int i = TOP; i >= 0; i--)
      cout << "\t|" << stack[i] << "|\n\t--" << endl;
  }
  void push (){
    TOP += 1;
    if (TOP == capacity){
	cout << "\tOverflow\n";
	TOP -= 1;
	return;
      }
    cout << "\tEnter element to insert in stack\n";
    int data;
    cout << "\t";
    cin >> data;
    stack[TOP] = data;
    cout << "\tSucessfully inserted\n";
  }
  int pop (){
    if (TOP == -1){
	cout << "\tUderflow \n";
	return 0;
      }
    int temp;
    temp = stack[TOP];
    stack[TOP] = 0;
    TOP -= 1;
    cout<<"\t"<<temp<<" poped from stack\n";
    return temp;
  }
  void stat ()
  {
    cout << "\tCapacity : " << capacity << endl;
    cout << "\tFilled : " << TOP+1 << endl;
    print();
  }

};

int main ()
{
  int size;
  cout << "\tEnter size of stack\n\t";
  cin >> size;
  Stack stack (size);
  system ("cls");
  while (1)
    {
      system ("cls");
      cout << "\tChoose the options\n";
      cout << "\t1. Push element in stack\n";
      cout << "\t2. Pop element from stack\n";
      cout << "\t3. Show current stats of stack\n";
      cout << "\t4. Exit\n";
      int options;
      cout << "\t";
      cin >> options;
      switch (options)
	{
	case 1:
	  stack.push ();
	  cout << "\t";getch();
	  break;
	case 2:
	  stack.pop ();
	  cout << "\t";getch();
	  break;
	case 3:
	  stack.stat ();
	  cout << "\t";getch();
	  break;
	case 4:
	  exit (0);
	default:
	  cout << "\tYou enterd wrong option\n";
	  cout << "press enter to continue....";
	  getch();
	}
    }
  return 0;
}
